spec/support/helpers.rb

```diff
@@ -1,32 +1,36 @@
 module SpecHelpers
-  def html_to_rgb(color)
-    generate_rgb_string(Color::RGB.from_html(color))
-  end
-
-  def html_to_rgb_for_string(string, base_color)
-    hash = Digest::SHA1.hexdigest string
-
-    hue_offset     = GeoPattern::PatternHelpers.map(GeoPattern::PatternHelpers.hex_val(hash, 14, 3), 0, 4095, 0, 359)
-    sat_offset     = GeoPattern::PatternHelpers.hex_val(hash, 17, 1)
-    base_color     = Color::RGB.from_html(base_color).to_hsl
-    base_color.hue = base_color.hue - hue_offset;
-
-    if (sat_offset % 2 == 0)
-      base_color.saturation = base_color.saturation + sat_offset
-    else
-      base_color.saturation = base_color.saturation - sat_offset
-    end
-
-    generate_rgb_string(base_color.to_rgb)
-  end
-
-  def generate_rgb_string(rgb)
-    r = (rgb.r * 255).round
-    g = (rgb.g * 255).round
-    b = (rgb.b * 255).round
-
-    format("rgb(%d, %d, %d)", r, g, b)
+  # Taken from active support
+  #
+  # Captures the given stream and returns it:
+  #
+  # stream = capture(:stdout) { puts 'notice' }
+  # stream # => "notice\n"
+  #
+  # stream = capture(:stderr) { warn 'error' }
+  # stream # => "error\n"
+  #
+  # even for subprocesses:
+  #
+  # stream = capture(:stdout) { system('echo notice') }
+  # stream # => "notice\n"
+  #
+  # stream = capture(:stderr) { system('echo error 1>&2') }
+  # stream # => "error\n"
+  def capture(stream)
```

Comment:
@ttaylorr it is, but this method is being deprecated due to it not being thread safe.

You can check it out here https://github.com/rails/rails/blob/346677b3137a35de65c660c230d6fe9e71aa9b0e/activesupport/lib/active_support/core_ext/kernel/reporting.rb

And here is the PR that talks about it https://github.com/rails/rails/pull/13392

Sorry for the random interjection I'm working a similar problem in `teachers_pet` :bowtie: